File: apps/CardHero.NetCoreApp.TypeScript/ClientApp/models/CardModel.ts
import { Rarity } from "../clients/clients";
import IMapper from "../utils/mapper";

export interface CardId extends Number {
    _cardIdBrand: number;
}

export default class CardModel implements IMapper<CardModel> {
    id: CardId;
    name: string;

    health: number;
    attack: number;
    upAttack: number;
    rightAttack: number;
    downAttack: number;
    leftAttack: number;
    defence: number;
    totalStats: number;

    rarity: Rarity;

    // available to logged in users
    isFavourited?: boolean;

    from(o?: any): CardModel {
        if (!o) return this;

        this.attack = o.attack;
        this.defence = o.defence;
        this.downAttack = o.downAttack;
        this.health = o.health;
        this.id = o.id;
        this.leftAttack = o.leftAttack;
        this.name = o.name;
        this.rightAttack = o.rightAttack;
        this.totalStats = o.totalStats;
        this.upAttack = o.upAttack;
        this.rarity = o.rarity;

        this.isFavourited = o.isFavourited;

        return this;
    }
}

File: src/CardHero.Core.Models/Models/CardModel.cs
namespace CardHero.Core.Models
{
    /// <summary>
    /// Card.
    /// </summary>
    public class CardModel : ICardHeroEntity
    {
        /// <summary>
        /// Id.
        /// </summary>
        public int Id { get; set; }

        /// <summary>
        /// Name.
        /// </summary>
        public string Name { get; set; }

        /// <summary>
        /// Description.
        /// </summary>
        public string Description { get; set; }

        /// <summary>
        /// Up attack.
        /// </summary>
        public int UpAttack { get; set; }

        /// <summary>
        /// Right attack.
        /// </summary>
        public int RightAttack { get; set; }

        /// <summary>
        /// Down attack.
        /// </summary>
        public int DownAttack { get; set; }

        /// <summary>
        /// Left attack.
        /// </summary>
        public int LeftAttack { get; set; }

        /// <summary>
        /// Health.
        /// </summary>
        public int Health { get; set; }

        /// <summary>
        /// Attack.
        /// </summary>
        public int Attack { get; set; }

        /// <summary>
        /// Defence.
        /// </summary>
        public int Defence { get; set; }

        /// <summary>
        /// Total stats.
        /// </summary>
        public int TotalStats { get; set; }

        /// <summary>
        /// Is favourited.
        /// </summary>
        public bool IsFavourited { get; set; }

        /// <summary>
        /// Rarity.
        /// </summary>
        public RarityModel Rarity { get; set; }
    }
}
